zebra-rpc/src/methods.rs

```diff
@@ -362,3 +459,44 @@ pub struct GetBlock(#[serde(with = "hex")] SerializedBlock);
 ///
 /// Also see the notes for the [`Rpc::get_best_block_hash` method].
 pub struct GetBestBlockHash(#[serde(with = "hex")] block::Hash);
+
+/// Response to a `getrawtransaction` RPC request.
+///
+/// See the notes for the [`Rpc::get_raw_transaction` method].
+// pub struct GetRawTransaction(pub(super) SerializedTransaction);
+#[derive(Clone, Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+pub enum GetRawTransaction {
+    /// The raw transaction, encoded as hex bytes.
+    Raw(#[serde(with = "hex")] Vec<u8>),
+    /// The transaction object.
+    Object {
+        /// The raw transaction, encoded as hex bytes.
+        #[serde(with = "hex")]
+        hex: Vec<u8>,
+        /// The height of the block that contains the transaction, or -1 if
+        /// not applicable.
+        height: i32,
+    },
+}
+
+impl GetRawTransaction {
+    fn from_transaction(
+        tx: Arc<Transaction>,
+        height: Option<block::Height>,
+        verbose: bool,
+    ) -> std::result::Result<Self, io::Error> {
+        let serialized_tx = tx.zcash_serialize_to_vec()?;
+        if verbose {
+            Ok(GetRawTransaction::Object {
+                hex: serialized_tx,
+                height: match height {
+                    Some(height) => height.0 as i32,
```

Comment:
Optional: `as` silently truncates data, but `try_into` returns an error.

Since these heights are from valid blocks in the state, this extra error handling is optional here. But either way we should document that expectation.

 I have a slight preference for `try_into`, because sometimes people copy code without reading the comments. ( Which wouldn't work for methods like `get_block` or `z_get_tree_state`. They handle arbitrary heights, so they can't use `as`.)

```suggestion
                    Some(height) => height.0.try_into().expect("valid block heights are limited to i32::MAX"),
```

Reply:
This was done.